### PR TITLE
ENG-2009: add install plan modal to ECR

### PR DESCRIPTION
**Pull request by gReis89**
This implements https://entando.myjetbrains.com/youtrack/issue/ENG-2009 and https://entando.myjetbrains.com/youtrack/issue/ENG-2042

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **greis89**

If they are not already assigned, you can assign the PR to them by writing `/assign @greis89` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/state/component-repository/components/reducer.js [hunk 1]
```diff
@@ -297,6 +300,75 @@ const progressStatus = (state = 0, action = {}) => {
   }
 };
 
+const initialConflictsModalState = {
```
Comment (gidesan) [inline src/state/component-repository/components/reducer.js:303]: Please remove "conflicts modal" with "install conflicts" (or whatever suits it better) within state. "Modal" is UI-related, not logic-related.

Consider this comment for all the occurrences of "conflicts modal" in the state files.

File: src/ui/component-repository/components/InstallationPlanModal.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  const version = useSelector(getComponentVersion);
+  const dispatch = useDispatch();
+  const [filterType, setFilterType] = useState('all');
+  const [requiredList, setRequiredList] = useState([]);
+
+  const componentList = [];
+
+  if (installPlan) {
+    Object.keys(installPlan)
+      .filter(key => !IGNORED_KEYS.includes(key))
```
Comment (gidesan) [inline src/ui/component-repository/components/InstallationPlanModal.js:35]: why not using `reduce` here, like you did in the reducer file?

File: src/ui/component-repository/components/InstallationPlanModal.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+export const MODAL_ID = 'InstallationPlanModal';
+const IGNORED_KEYS = ['version', 'conflictStrategy', 'hasConflicts'];
+const ACTIONS = {
+  SKIP: 'skip',
+  OVERRIDE: 'update',
+  CREATE: 'install',
+};
+
+const InstallationPlanModal = ({
```
Comment (gidesan) [inline src/ui/component-repository/components/InstallationPlanModal.js:21]: it's a 150+ LoC component....can we split it?

**Review by gidesan — CHANGES_REQUESTED** [review]

File: src/ui/component-repository/components/InstallationPlanModal.js (hunk 2 above)
Comment (gReis89) [reply src/ui/component-repository/components/InstallationPlanModal.js:35]: in the reducer file, I'm keeping the object structure, and applying an update, here I'm converting the two dimension object into a one dimension array. I didn't thought on using reduce for that but I think the code might be bigger, let me try.

File: src/ui/component-repository/components/InstallationPlanModal.js (hunk 3 above)
Comment (gReis89) [reply src/ui/component-repository/components/InstallationPlanModal.js:21]: yep, I think I can move the table to its own component, let me see

Comment (gReis89) [reply src/ui/component-repository/components/InstallationPlanModal.js]: Created a new InstallationPlanTable and split the code accordingly, now this one has around 100 lines.

File: src/ui/component-repository/components/InstallationPlanModal.js (hunk 2 above)
Comment (gReis89) [reply src/ui/component-repository/components/InstallationPlanModal.js]: I was able to change the implementation to use reduce and map and kept the same amount of line 👍

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1034#pullrequestreview-621082369" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**